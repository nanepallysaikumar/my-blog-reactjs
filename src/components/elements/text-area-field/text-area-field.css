.text-area-field {
    height: 200px;
    width: 100%;
}

.text-area-field.error {
    border: 1px solid #d64161;
}
