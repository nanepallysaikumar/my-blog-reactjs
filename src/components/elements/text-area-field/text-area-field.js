import React from 'react';
import classnames from 'classnames';
import { FormField } from '../form-field/form-field';

import './text-area-field.css';

export const TextArea = props => (
    <FormField title={props.title} error={props.error} touched={props.touched}>
        <textarea
            placeholder={props.placeholder}
            {...props.field}
            className={classnames('text-area-field', {
                error: props.error && props.touched
            })}
        />
    </FormField>
);
