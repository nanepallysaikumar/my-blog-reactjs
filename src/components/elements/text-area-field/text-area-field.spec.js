import { render, cleanup } from 'react-testing-library';
import React from 'react';
import { TextArea } from './text-area-field';

const setup = (touched, error) => {
    return render(
        <TextArea
            title="Test Title"
            field={{
                name: 'test',
                value: 'test',
                onChange: jest.fn()
            }}
            touched={touched}
            error={error}
        />
    );
};

describe('text-area-field', () => {
    it('should render normal text-area', async () => {
        const { container } = setup(false, undefined);
        expect(container).toMatchSnapshot('text-area-normal-snapshot');
    });

    it('should render error text-area', async () => {
        const { container } = setup(true, 'Invalid');
        expect(container).toMatchSnapshot('text-area-error-snapshot');
    });

    afterEach(cleanup);
});
