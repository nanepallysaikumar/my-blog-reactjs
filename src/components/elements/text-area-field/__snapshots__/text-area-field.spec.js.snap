// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-area-field should render error text-area: text-area-error-snapshot 1`] = `
<div>
  <div
    class="form-field"
  >
    <label
      class="form-field-label"
    >
      Test Title
    </label>
    <textarea
      class="text-area-field error"
      name="test"
    >
      test
    </textarea>
    <div
      class="form-field-error"
      data-testid="form-field-error-undefined"
    >
      Invalid
    </div>
  </div>
</div>
`;

exports[`text-area-field should render normal text-area: text-area-normal-snapshot 1`] = `
<div>
  <div
    class="form-field"
  >
    <label
      class="form-field-label"
    >
      Test Title
    </label>
    <textarea
      class="text-area-field"
      name="test"
    >
      test
    </textarea>
  </div>
</div>
`;
